sdk/core/core-http/src/policies/deserializationPolicy.ts

```diff
@@ -134,7 +134,7 @@ function shouldDeserializeResponse(parsedResponse: HttpOperationResponse): boole
   return result;
 }
 
-export function deserializeResponseBody(
+export async function deserializeResponseBody(
```

Comment:
curious what rule cause us  to change this to an async method.

Reply:
This function does have `Promise<HttpOperationResponse>` as the return type though. I think we can assume that it made sense to me to add the async there for consistency, as in: I would add it again if I encountered this without knowing your comment. However, I really don't remember why I did this. I'll put it back.